diff -urN a/bazel/private/upb_proto_library_internal/aspect.bzl b/bazel/private/upb_proto_library_internal/aspect.bzl
--- a/bazel/private/upb_proto_library_internal/aspect.bzl	2026-07-31 18:47:32
+++ b/bazel/private/upb_proto_library_internal/aspect.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """Implementation of the aspect that powers the upb_*_proto_library() rules."""
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel/common:proto_common.bzl", "proto_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load(":upb_proto_library_internal/cc_library_func.bzl", "cc_library_func")
diff -urN a/bazel/private/upb_proto_library_internal/cc_library_func.bzl b/bazel/private/upb_proto_library_internal/cc_library_func.bzl
--- a/bazel/private/upb_proto_library_internal/cc_library_func.bzl	2026-07-31 18:47:32
+++ b/bazel/private/upb_proto_library_internal/cc_library_func.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """A function to compile C/C++ code, like cc_library() but from Starlark."""
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain", "use_cpp_toolchain")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 def upb_use_cpp_toolchain():
     return use_cpp_toolchain()
diff -urN a/bazel/upb_c_proto_library.bzl b/bazel/upb_c_proto_library.bzl
--- a/bazel/upb_c_proto_library.bzl	2026-07-31 18:47:32
+++ b/bazel/upb_c_proto_library.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """upb_c_proto_library() exposes upb's generated C API for protobuf (foo.upb.h)"""
 
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load("//bazel/private:upb_proto_library_internal/aspect.bzl", "upb_proto_aspect_impl")
 load("//bazel/private:upb_proto_library_internal/cc_library_func.bzl", "upb_use_cpp_toolchain")
diff -urN a/bazel/upb_minitable_proto_library.bzl b/bazel/upb_minitable_proto_library.bzl
--- a/bazel/upb_minitable_proto_library.bzl	2026-07-31 18:47:32
+++ b/bazel/upb_minitable_proto_library.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """upb_minitable_proto_library() exposes upb's generated minitables (foo.upb_minitable.h)"""
 
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel/private:upb_proto_library_internal/aspect.bzl", "upb_proto_aspect_impl")
 load("//bazel/private:upb_proto_library_internal/cc_library_func.bzl", "upb_use_cpp_toolchain")
 load("//bazel/private:upb_proto_library_internal/rule.bzl", "upb_proto_rule_impl")
diff -urN a/bazel/upb_proto_reflection_library.bzl b/bazel/upb_proto_reflection_library.bzl
--- a/bazel/upb_proto_reflection_library.bzl	2026-07-31 18:47:32
+++ b/bazel/upb_proto_reflection_library.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """upb_c_proto_reflection_library() exposes upb reflection for protobuf (foo.upbdefs.h)"""
 
 load("//bazel:upb_minitable_proto_library.bzl", "UpbMinitableCcInfo", "upb_minitable_proto_library_aspect")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 load("//bazel/common:proto_common.bzl", "proto_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load("//bazel/private:upb_proto_library_internal/aspect.bzl", "upb_proto_aspect_impl")
diff -urN a/build_defs/cc_proto_blacklist_test.bzl b/build_defs/cc_proto_blacklist_test.bzl
--- a/build_defs/cc_proto_blacklist_test.bzl	2026-07-31 18:47:32
+++ b/build_defs/cc_proto_blacklist_test.bzl	2026-07-31 18:47:32
@@ -1,6 +1,7 @@
 """Contains a unittest to verify that `cc_proto_library` does not generate code for blacklisted `.proto` sources (i.e. WKPs)."""
 
 load("@bazel_skylib//lib:unittest.bzl", "asserts", "unittest")
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
 
 def _cc_proto_blacklist_test_impl(ctx):
     """Verifies that there are no C++ compile actions for Well-Known-Protos.
